.github/workflows/test_python.yml

```diff
@@ -0,0 +1,43 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Pytest and Lint
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.11"]
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --editable .
+        python -m pip install --upgrade pip
+        python -m pip install flake8 pytest
+    - name: Set Env variable
+      run: |
+        export AEIC_DATA_DIR=$(pwd)/data
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```

Comment:
I'd suggest running the linter as a separate job (it can be part of the same workflow) so you can see the status for this and the tests individually.

I'd also advocate for using the Black/Ruff default of 88 characters per line as the max.

I think the linter config should be done via the `pyproject.toml`, to facilitate running the linter locally.

You may also want to consider Ruff instead of flake8.